### PR TITLE
Listen for chain node start before connecting

### PR DESCRIPTION
**Pull request by maael**
Changes so we don't continuously ping to see if it's up and we can connect, instead we wait for the process to tell us the node has started.

🚨 Need to decide how to handle failure when connecting @MetalMichael ? 🚨

### CODE REVIEW

File: EvotoClient/Blockchain/MultiChainHandler.cs [hunk 1]
```diff
@@ -105,14 +92,23 @@ public static string RandomString(int length)
...
+
+        private Task RunDaemon(Func<Task> successCallback)
         {
             if (_process != null)
             {
                 if (_process.HasExited)
                     Debug.WriteLine("Restarting Multichain!!");
                 else
-                    return true;
+                    return successCallback();
```
Comment (MetalMichael) [inline EvotoClient/Blockchain/MultiChainHandler.cs]: A few of these should be awaiting the task, not returning a reference to them. Method should also be async.

**Comment by MetalMichael** [comment]
Failure shouldn't matter too much, as it already clears the appdata folder, so _shouldn't_ happen. Maybe just throw a warning and exit. As it's only prototype.

**Comment by maael** [comment]
@MetalMichael Changed it so that it throws a warning exception which will be caught and logged in `Connect`. So the connection fails, but the app is left running. Cool with that?

**Comment by MetalMichael** [comment]
@maael sure, potentially make a custom exception for it, and not catch Exception though, as it's too generic.

**Review by MetalMichael — CHANGES_REQUESTED** [review]

**Comment by maael** [comment]
I don't see a problem with catching all exceptions in connect at the moment. We don't want anything to be thrown and kill the gui. Why should it be `async`, you only need `async` as a heads up that you're going to `await` right it doesn't actually change anything besides allow optimisation for `await`. Why do we need to `await` the success callback anywhere?

**Comment by MetalMichael** [comment]
Because otherwise the reference to the Task is passed around, rather than just awaiting it where it's called, which is better practice. Otherwise you could potentially miss awaiting the task, and end up with weird sync issues. 

Just seems unusual having a method return a task that isn't async. Not necessarily wrong, but not intuitive.

image

**Comment by maael** [comment]
Thing being left over from javascript I guess, I'm used to passing functions around everywhere.

**Review by MetalMichael — APPROVED** [review]

**Pull request merged**